pkg/prometheus/resource_selector.go

```diff
@@ -42,6 +42,16 @@ import (
 	"github.com/prometheus-operator/prometheus-operator/pkg/operator"
 )
 
+// The valid options for Role.
+const (
```

Comment:
it would be good to have the consts defined in pkg/apis/monitoring/v1 directly.

Reply:
@simonpasquier previously i have done the same, but @slashpai suggested me to put it in the resource_selector

Reply:
Thats for validation methods 🙂
Since consts were also moved I was ok but I agree with Simon consts go better in types.go

Reply:
Let's have a follow-up for this.